nixops/backends/__init__.py

```diff
@@ -455,15 +455,15 @@ def get_scp_name(self):
         return ssh_name
 
     def upload_file(self, source, target, recursive=False):
-        master = self.ssh.get_master()
+        master = self.ssh.get_master(user=self.ssh_user)
```

Comment:
Let's refactor the `get_master` stuff so call-sites don't need to remember to pass `user=...`. From what we were discussing, it sounds like `get_master` should be a private function anyway, internal to the SSH util.

Reply:
I would like to do that in a follow-up PR. This PR is already big enough and it might require some significant refactoring.

Reply:
re #1307